majifix311/src/main/java/com/example/majifix311/ui/adapters/OpenClosedTabAdapter.java

```
@@ -0,0 +1,94 @@
+package com.example.majifix311.ui.adapters;
+
+import android.content.Context;
+import android.support.v4.app.Fragment;
+import android.support.v4.app.FragmentManager;
+import android.support.v4.app.FragmentStatePagerAdapter;
+
+import com.example.majifix311.R;
+import com.example.majifix311.models.Problem;
+import com.example.majifix311.ui.ProblemListFragment;
+import com.example.majifix311.utils.ProblemUtils;
+
+import java.util.ArrayList;
+
+public class OpenClosedTabAdapter extends FragmentStatePagerAdapter {
+    private static int NUM_ITEMS = 3;
+
+    private ArrayList<Problem> all;
+    private ArrayList<Problem> open;
+    private ArrayList<Problem> closed;
+
+    private final Context mContext;
+
+    private boolean isEmpty;
+
+    public OpenClosedTabAdapter(
+            Context ctx, FragmentManager fm, ArrayList<Problem> requests) {
+        super(fm);
+
+        this.mContext = ctx;
+
+        if (requests.isEmpty()) {
+            isEmpty = true;
+            return;
+        }
+
+//        This can be used to mock a CLOSED request for testing purposes on list view
+//        ServiceRequest mockRequest = new Problem();
+//        mockRequest.description = "mock desc";
+//        mockRequest.service = new Service("B", "Billing", "#FFFFFF");
+//        mockRequest.createdAt = new Date(2017, 10, 3, 12, 19);
+//        mockRequest.status = new Status(CLOSED, "#000000");
+//        requests.add(mockRequest);
+
+        all = new ArrayList<>();
+        open = new ArrayList<>();
+        closed = new ArrayList<>();
+
+        ProblemUtils.sortByDate(all);
+        for (Problem request : requests) {
+            if (request != null) {
```

Probably shouldn't be addressed here, but should we view a null Problem being passed around as a violation of our internal contract?

(For reference, I buy into the philosophy that while code should be resilient to user & external interaction, it should be brittle (fail fast) with regard to violation of contract, e.g. malformed digital data. If we don't want to proceed as such, this comment is moot.)